pages/k8s/upgrading.md

```diff
@@ -0,0 +1,319 @@
+---
+
+# Upgrading
+
+It is recommended that you keep your **Kubernetes** deployment updated to the
+latest available stable version. You should also update the other applications
+which make up the **Canonical Distribution of Kubernetes<sup>&reg;</sup>.**
+Keeping up to date makes sure you have the latest bug-fixes and security
+patches for smooth operation of your cluster.
+
+There is no set cadence to the releases of **Kubernetes**,  but you can check
+the latest release version on the  
+[Kubernetes release page on GitHub][k8s-release].  The **CDK** is kept in close sync
+with upstream Kubernetes.
+
+<div class="p-notification--information">
+  <p class="p-notification__response">
+    <span class="p-notification__status">Note:</span>
+<strong>Kubernetes</strong> will automatically handle patch releases. This means that
+the cluster will perform an unattended automatic upgrade between patch versions, e.g.
+1.10.7 to 1.10.8. Attended upgrades are only required when you wish to upgrade a minor
+version, e.g. 1.9.x to 1.10.x.
+  </p>
+</div>
+
+You can see which version of each application is currently deployed by running
+
+```bash
+juju status
+```
+
+The 'App' section of the output lists each application and its version number.
+Note that this is the version of the upstream application deployed. The version
+of the Juju charm is indicated under the column titled 'Rev'. The charms may be
+updated in between new versions of the application.
+
+## Before you begin
+
+As with all upgrades, there is a possibility that there may be unforeseen
+difficulties. It is **highly recommended that you make a backup** of any
+important data, including any running workloads. For more details on creating
+backups, see the separate [documentation on backups][backups].
+
+You should also make sure:
+
+  * The machine from which you will perform the backup has sufficient internet access to
+      retrieve updated software
+  * Your cluster is running normally
+  * You read the [Upgrade notes][notes] to see if any caveats apply to the versions you are
+     upgrading to/from
+  * You read the [Release notes][release-notes] for the version you are upgrading to,
+      which will alert you to any important changes to the operation of your cluster
+
+## Infrastructure updates
+
+The applications which run alongside the core Kubernetes components can be
+upgraded at any time. These applications are widely used and may frequently
+receive upgrades outside of the cycle of new releases of Kubernetes.
+
+This includes:
+
+ - easyrsa  
+ - etcd                
+ - flannel
+
+ Note that this may include other applications which you may have installed, such as
+ Elasticsearch, Prometheus, Nagios, Helm, etc.
+
+### Upgrading **etcd**
+
+ As **etcd** manages critical data for the cluster, it is advisable to create a snapshot of
+ this data before running an upgrade. This is covered in more detail in the  
```

Comment:
The upstream instructions say you MUST create a snapshot, but doesn't then mention anything about restoring from the snapshot afterwards. I have included the instructions here, but it seems that actually, this is just something you **should** do